openmole/plugins/org.openmole.plugin.tool.pattern/src/main/scala/org/openmole/plugin/tool/pattern/Strain.scala

```diff
@@ -28,7 +29,14 @@ object Strain {
     val first = Capsule(EmptyTask(), strainer = true)
     val last = Slot(Capsule(EmptyTask(), strainer = true))
 
-    (first -- puzzle -- last) + (first -- last)
+    val mole = puzzle.toMole
+    val sources = Sources(puzzle.sources.groupBy(_._1).mapValues(_.map(_._2)))
+    val hooks = Hooks(puzzle.hooks.groupBy(_._1).mapValues(_.map(_._2)))
```

Comment:
> The best place would be to write the implicit conversion in the Sources and Hooks companion objects (uncreated yet)?

I think the companion objects already exist, just at the bottom not right next to the classes... will move them up, too.